var/spack/repos/builtin/packages/precice/package.py

```diff
@@ -195,7 +195,7 @@ def cmake_args(self):
             python_library = spec["python"].libs[0]
             python_include = spec["python"].headers.directories[0]
             numpy_include = join_path(
-                spec["py-numpy"].package.module.python_platlib, "numpy", "core", "include"
+                self["py-numpy"].module.python_platlib, "numpy", "core", "include"
```

Comment:
Could you explain why we need to use `self` instead of `spec` here? I haven't come across this by now.

Reply:
We don't need, it's just shorter, since we want to use a property of the package object anyhow. This is possible since https://github.com/spack/spack/pull/48467 was merged.

Reply:
Basically, we are trying to separate the responsibility of specs and packages better, and #48467 allows to traverse the packages with the same syntax as spec, so idioms like:
```python
self.spec["foo"].package.bar()
```
are not necessary anymore, since one can write:
```python
self["foo"].bar()
```

Reply:
Ah nice! Thanks for the quick reply.